js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -96,18 +107,63 @@ export function buildAssetCountBySection(assets: AssetDefinitionMetadata[]): Ass
         (assetCountByCodeLocation[stringifiedCodeLocation] || 0) + 1;
     });
 
-  const countPerAssetGroup = Object.entries(assetCountByGroup).map(([groupIdentifier, count]) => ({
-    assetCount: count,
-    groupMetadata: JSON.parse(groupIdentifier),
-  }));
+  const countsByOwner = Object.entries(assetCountByOwner)
+    .map(([owner, count]) => ({
+      owner,
+      assetCount: count,
+    }))
+    .sort(({owner: ownerA}, {owner: ownerB}) => COMMON_COLLATOR.compare(ownerA, ownerB));
+  const countsByComputeKind = Object.entries(assetCountByComputeKind)
+    .map(([computeKind, count]) => ({
+      computeKind,
+      assetCount: count,
+    }))
+    .sort(({computeKind: computeKindA}, {computeKind: computeKindB}) =>
+      COMMON_COLLATOR.compare(computeKindA, computeKindB),
+    );
+  const countPerAssetGroup = Object.entries(assetCountByGroup)
+    .map(([groupIdentifier, count]) => ({
+      assetCount: count,
+      groupMetadata: JSON.parse(groupIdentifier),
+    }))
+    .sort(
+      (
+        {
+          groupMetadata: {
+            groupName: groupNameA,
+            repositoryLocationName: repositoryLocationNameA,
+            repositoryName: repositoryNameA,
+          },
+        },
+        {
+          groupMetadata: {
+            groupName: groupNameB,
+            repositoryLocationName: repositoryLocationNameB,
+            repositoryName: repositoryNameB,
+          },
+        },
+      ) =>
+        COMMON_COLLATOR.compare(repositoryLocationNameA, repositoryLocationNameB) ||
+        COMMON_COLLATOR.compare(repositoryNameA, repositoryNameB) ||
+        COMMON_COLLATOR.compare(groupNameA, groupNameB),
+    );
+  const countPerCodeLocation = Object.entries(assetCountByCodeLocation)
+    .map(([key, count]) => ({
+      repoAddress: repoAddressFromPath(key)!,
+      assetCount: count,
+    }))
+    .sort(
+      (
+        {repoAddress: {name: nameA, location: locationA}},
+        {repoAddress: {name: nameB, location: locationB}},
+      ) => COMMON_COLLATOR.compare(locationA, locationB) || COMMON_COLLATOR.compare(nameA, nameB),
```

Comment:
Probably best to use `repoAddressAsHumanString` for this, since the location can often come after the `@`, plus we don't get value out of doing a string comparison on the dunder repo name.

Reply:
Refactored to that!